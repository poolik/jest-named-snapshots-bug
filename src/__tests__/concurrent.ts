
describe('My tests', () => {

  describe('first group', () => {
    it.concurrent('first', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'first'};
      expect(result).toMatchSnapshot('first')
    });

    it.concurrent('second', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'second'};
      expect(result).toMatchSnapshot('second')
    });

    it.concurrent('third', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'third'};
      expect(result).toMatchSnapshot('third')
    });

    it.concurrent('fourth', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'fourth'};
      expect(result).toMatchSnapshot('fourth')
    });

    it.concurrent('fifth', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'fifth'};
      expect(result).toMatchSnapshot('fifth')
    });
  });

  describe('second group', () => {
    it.concurrent('first', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'first', group:2};
      expect(result).toMatchSnapshot('2_first')
    });

    it.concurrent('second', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'second', group:2};
      expect(result).toMatchSnapshot('2_second')
    });

    it.concurrent('third', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'third', group:2};
      expect(result).toMatchSnapshot('2_third')
    });

    it.concurrent('fourth', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'fourth', group:2};
      expect(result).toMatchSnapshot('2_fourth')
    });

    it.concurrent('fifth', async () => {
      await wait(getRandomInt(100, 1000));
      const result = {data: 'fifth', group:2};
      expect(result).toMatchSnapshot('2_fifth')
    });
  });

});

function getRandomInt(min, max) {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

function wait(time: number) {
  return new Promise((resolve) => {
    setTimeout(() => { // simulate a server delay
      resolve();
    }, time);
  });
}