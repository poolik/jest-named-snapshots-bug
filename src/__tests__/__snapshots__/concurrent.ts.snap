// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My tests first group first: 2_first 1`] = `
Object {
  "data": "first",
  "group": 2,
}
`;

exports[`My tests first group first: 2_second 1`] = `
Object {
  "data": "second",
  "group": 2,
}
`;

exports[`My tests first group first: fifth 1`] = `
Object {
  "data": "fifth",
}
`;

exports[`My tests first group first: first 1`] = `
Object {
  "data": "first",
}
`;

exports[`My tests first group first: third 1`] = `
Object {
  "data": "third",
}
`;

exports[`My tests first group fourth: 2_third 1`] = `
Object {
  "data": "third",
  "group": 2,
}
`;

exports[`My tests first group fourth: fourth 1`] = `
Object {
  "data": "fourth",
}
`;

exports[`My tests first group second: 2_fifth 1`] = `
Object {
  "data": "fifth",
  "group": 2,
}
`;

exports[`My tests first group second: 2_fourth 1`] = `
Object {
  "data": "fourth",
  "group": 2,
}
`;

exports[`My tests first group second: second 1`] = `
Object {
  "data": "second",
}
`;
